src/server.js

```diff
@@ -365,11 +367,11 @@ function initializeClient(socket, client, generateToken, token) {
 				}
 			});
 
-			sendInitEvent(token);
-		});
-	} else {
-		sendInitEvent(null);
+			return newToken;
+		}).catch((error) => log.error(error));
 	}
+
+	Promise.resolve(tokenPromise).then((t) => sendInitEvent(t));
```

Comment:
You could move `sendInitEvent` code here.

Reply:
Done.